lib/crash_reporter.js

```diff
@@ -0,0 +1,191 @@
+var fs = require('fs');
+var logs = require('./logs');
+var packageJson = require('../package.json');
+var path = require('path');
+var request = require('request');
+
+// the value of the crash reporter preference
+// the value has to one of 'on' or 'off'
+var CRASH_REPORTER_PREFERENCE = 'crash.reporter.preference';
+var SUBMIT_CRASH_URL = 'http://tessel-error-reporter.appspot.com/crashes/submit';
+
+// override for testing
+// SUBMIT_CRASH_URL = 'http://localhost:8080/crashes/submit';
+
+var preferenceFile = path.join(__dirname, '__preferences.json');
+
+var Preferences = function() {
+
+};
+
+Preferences.prototype.read = function(key, defaultValue) {
```

Comment:
Structurally, it seems like the `Preferences` class should be in its own file since it's not specific to crash reporting.